test/function.js

```diff
@@ -0,0 +1,88 @@
+"use strict";
+
+const assert = require("assert");
+
+const conversions = require("..");
+
+const supportsAsyncFunction = (() => {
+    try {
+        eval("(async function () {})");
+        return true;
+    } catch (err) {
+        return false;
+    }
+})();
+
+function test(type) {
+    describe("WebIDL " + type + " type", () => {
+        const sut = conversions[type];
+
+        it("should return `function () {}` for `function () {}`", () => {
+            const func = function () {};
+            assert.strictEqual(sut(func), func);
+        });
+
+        it("should return `function* () {}` for `function* () {}`", () => {
+            const func = function* () {};
+            assert.strictEqual(sut(func), func);
+        });
+
+        it("should return `() => {}` for `() => {}`", () => {
+            const func = () => {};
+            assert.strictEqual(sut(func), func);
+        });
+
+        it("should return `async function () {}` for `async function () {}`", supportsAsyncFunction ? () => {
```

Comment:
I'd rather wrap these two in `if()` blocks than use the conditional inside the argument to `it()`.